### ISSUE TITLE
Prosthetic and Augmented Replacement Arms Don't Work

### ISSUE THREAD

**Issue by JustAnotherUserGH** [tgstation/tgstation#20457]
A recent round involved a lot of places around the station getting bombed. As one of the victims, my character got a limb blown off. However, even after the arm was reattached, it was nonfunctional. I tried getting it augmented - no change. As a doctor, I helped other people who'd been caught in the mayhem - replacing a janitor's arm straight away with a cyborg limb led to him having the same problem.

Clearly, arm replacements should work - otherwise, what's the point of having them? Nothing else extraordinary happened - getting my arm blown off was one of the first parts of the round, it was with a model ship bomb. Picked up the arm from the breach, went to get it put back on.

Was playing on Sybil, Box was the Map. Server revision "ce298aa381eea52140b67f21ac27605695153dcf" with test merges of "#20441" and "#20445".

**Comment by Jalleo** [comment]
@phil235

**Comment by MrStonedOne** [comment]
@RemieRichards

**Comment by RemieRichards** [comment]
When an arm is attached it's probably not adding itself to ```hand_bodyparts```, which is used to valid held indexes, which is why the arms aren't working.

**Comment by ExcessiveUseOfCobblestone** [comment]
When you replace your arm with surgery, it creates a new bullet under `hand_bodyparts` instead of replacing it under the second bullet in the list [if that makes sense].

I fixed it ingame by just clearing nulls.

**Comment by phil235** [comment]
@ExcessiveUseOfCobblestone I think it might be due to has_right_hand() and has_left_hand() being swapped. (I made a fix for it in dismemberment PR)

**Comment by Jalleo** [comment]
Requires testing then?

**Comment by Incoming5643** [comment]
This is a pretty major bug guys

**Comment by TehZombehz** [comment]
Can the PR that was supposed to fix this be re-addressed? I've had to do prosthetic surgery on one too many people just to find that their arms still don't work. Legs still seem fine, but reattaching an arm doesn't allow it to be used, and I even proceeded to augment the arm afterwards and it still didn't fix it. I've actually had to clone somebody to fix their limbs because the normal method won't work. It's kind of a big problem.
